trackers/run_tracker.py

```diff
@@ -72,26 +85,62 @@ def run(global_config_file, dash_schema_file, pipelines, run_id=1):
                         status = func(subject_dir, session_id, run_id)
                         # print(f"task_name: {name}, status: {status}")
                         _df.loc[bids_id,name] = status
-                        _df.loc[bids_id,"pipeline_starttime"] = get_start_time(subject_dir)
-                        _df.loc[bids_id,"pipeline_endtime"] = UNAVAILABLE # TODO
+                    _df.loc[bids_id,"pipeline_starttime"] = get_start_time(subject_dir)
+                    _df.loc[bids_id,"pipeline_endtime"] = get_end_time(subject_dir)
+                    n_participants += 1
                 else:
                     # print(f"Pipeline output not found for bids_id: {bids_id}, session: {session}")
                     for name in status_check_dict.keys():                    
                         _df.loc[bids_id,name] = UNAVAILABLE
-                        _df.loc[bids_id,"pipeline_starttime"] = UNAVAILABLE
-                        _df.loc[bids_id,"pipeline_endtime"] = UNAVAILABLE
+                    _df.loc[bids_id,"pipeline_starttime"] = UNAVAILABLE
+                    _df.loc[bids_id,"pipeline_endtime"] = UNAVAILABLE
+
+                # don't check all participants if testing
+                if testing and n_participants > N_TESTING:
+                    break
+
+            proc_status_session_dfs.append(_df)
 
-            proc_status_dfs.append(_df)
+        # new rows for this pipeline
+        pipeline_proc_status_df = pd.concat(proc_status_session_dfs, axis='index').reset_index(drop=True)
 
-    proc_status_df = pd.concat(proc_status_dfs, axis='index')
+        # add old rows from other pipelines and sort for consistent order
+        proc_status_df: pd.DataFrame = pd.concat([old_proc_status_df, pipeline_proc_status_df], axis='index')
+        proc_status_df = proc_status_df.sort_values(["pipeline_name", "pipeline_version", "bids_id"], ignore_index=True)
 
-    # Save proc_status_df
-    tracker_csv = f"{mr_proc_root_dir}/derivatives/bagel.csv"
-    proc_status_df = proc_status_df.drop(columns="bids_id")
-    proc_status_df.index.name = "bids_id"
-    proc_status_df.to_csv(tracker_csv)
+        # don't write a new file if no changes
+        try:
+            if len(proc_status_df.compare(old_proc_status_df_full)) == 0:
+                print(f'\nNo change for pipeline {pipeline}')
```

Comment:
I added the check for the participant list: if the participant list from the manifest isn't the same as the one from the existing bagel and if the pipelines in the existing bagel have not all been specified, raise error and tell the user to rerun with `--pipelines <all_the_pipelines>`.

Interesting idea for the extra subject list argument. To avoid having to specify the list of participants, maybe we can use a `--new-only` flag so that it only runs the tracker for new subject-session pairs in the manifest? @nikhil153 what do you think?

Reply:
Okay, all interesting ideas! I think it would be helpful to come up with scenarios / use-cases where these options will be applicable (also helps documentation and upcoming demo!).
Generally, in the context of mr_proc, I prefer not having custom subject-lists with ability to update bagels in ad-hoc fashion. It could create confusion. At the same time, there are scenarios where we would want to re-run batch of subjects with new pipeline arguments to fix something... And I am not sure if `--new-only` flag or any other easy workaround would allow us to update selective bagels rows.